Dockerfile

```diff
@@ -0,0 +1,56 @@
+FROM debian:stable as builder
+
+RUN apt-get update && apt-get install -y --no-install-recommends curl git wget ca-certificates
+
+WORKDIR /java/
+
+# Download the JRE for copying to the image to run the OTP server
+RUN curl -Lo jre17.tar.gz https://github.com/adoptium/temurin17-binaries/releases/download/jdk-17.0.7%2B7/OpenJDK17U-jre_x64_linux_hotspot_17.0.7_7.tar.gz
+RUN tar xvf jre17.tar.gz && rm jre17.tar.gz
+
+# Download the JDK and maven and add them to path for building OTP and onebusaway-gtfs-modules
+RUN curl -Lo jdk17.tar.gz https://github.com/adoptium/temurin17-binaries/releases/download/jdk-17.0.7%2B7/OpenJDK17U-jdk_x64_linux_hotspot_17.0.7_7.tar.gz
+RUN tar xvf jdk17.tar.gz && rm jdk17.tar.gz
+ENV JAVA_HOME=/java/jdk-17.0.7+7/
+ENV PATH="$JAVA_HOME/bin:$PATH"
+
+RUN curl -Lo maven.tar.gz https://dlcdn.apache.org/maven/maven-3/3.9.1/binaries/apache-maven-3.9.1-bin.tar.gz
+RUN tar xvf maven.tar.gz && rm maven.tar.gz
+ENV PATH="/java/apache-maven-3.9.1/bin/:$PATH"
+
+# Build the MBTA branch of onebusaway-gtfs-modules so that it can be used in the OTP build
+WORKDIR /build/onebusaway-gtfs-modules
+# TODO: Remove the specific branch once the upgrade branch is merged into master
+RUN git clone --depth 1 -b otp2-upgrade --single-branch https://github.com/mbta/onebusaway-gtfs-modules .
+RUN ./build.sh
+
+# Download the latest GTFS and PBF files, then build OTP
+WORKDIR /build/OpenTripPlanner
+COPY . .
+ARG MBTA_GTFS_URL=https://mbta-gtfs-s3.s3.amazonaws.com/google_transit.zip
+RUN MBTA_GTFS_URL="$MBTA_GTFS_URL" ./mbta/update_gtfs.sh
+RUN ./mbta/update_pbf.sh
+RUN ./mbta/build.sh
+
+FROM debian:stable-slim as runner
+RUN apt-get update && apt-get upgrade -y && apt-get install -y --no-install-recommends dumb-init
+
+RUN useradd -MU otp
+USER otp
+
+# Copy the OTP JAR and build artifacts into the final image
+COPY --from=builder --chown=otp:otp /build/OpenTripPlanner/otp-2.3.0*-shaded.jar /dist/OpenTripPlanner/
+COPY --from=builder --chown=otp:otp /build/OpenTripPlanner/var/graph.obj /dist/OpenTripPlanner/var/
+COPY --from=builder --chown=otp:otp /build/OpenTripPlanner/var/*.json /dist/OpenTripPlanner/var/
+COPY --from=builder --chown=otp:otp /java/jdk-17.0.7+7-jre /java/jdk-17.0.7+7-jre
+
+# Set the default java install to the JRE that was copied into the image rather than the JDK
+ENV JAVA_HOME="/java/jdk-17.0.7+7-jre"
+ENV PATH="$JAVA_HOME/bin:$PATH"
+
+EXPOSE 5000
+
+# Run the OTP server, exposed on port 5000
+WORKDIR /dist/OpenTripPlanner
+ENTRYPOINT ["/usr/bin/dumb-init", "--"]
+CMD ["java", "-Xmx6G", "-jar", "otp-2.3.0-MBTA-shaded.jar", "--load", "var/", "--port", "5000"]
```

Comment:
Is there a way to define the port number in a single location? Currently, it is duplicated both here and on line 51

Reply:
I believe it would be beneficial to add a new line at the end of the file

Reply:
Additionally, do we need to specify the exact version (2.3.0) everywhere? By doing so, we would have to update all these scripts again if we decide to update the version. Perhaps we can simply refer to it as otp-MBTA-shaded.jar.